Shopify/ruby-lsp
# Documentation: make sure referenced GIF link exists

### Motivation

Closes: #684 

Make sure GIFs documentation really exists in check_docs task.

### Implementation

- Pass to the rake task the GIF paths list
- From path request, extract related GIF ex: `lib/ruby_lsp/requests/hover.rb` search for `hover.gif` in GIF paths list

Notes: 
- This test requires matching name between request and GIF (`inlay_hints` for example was without `s`), but I think it's acceptable. As always, you can challenge this choice!
- `PathCompletion` was missing, I added it to avoid red CI.    

### Automated Tests

This task is not tested (or I missed it, feel free to correct me).

### Manual Tests

- checkout this branch
- remove one GIF
- run `bundle exec rake ruby_lsp:check_docs`
- rake task should fail
